package.json

```diff
@@ -41,6 +41,9 @@
     "react-test-renderer": "16.13.1",
     "metro-react-native-babel-preset": "^0.61.0"
   },
+  "peerDependencies": {
+    "@react-native-async-storage/async-storage": "^1.15.5"
```

Comment:
I've never used `pperDependencies` before. What is it for and is it necessary?

Reply:
I've put that into the description but here's some more information 

`AsyncStorage` is linked to native `ios/android` through the project that uses it - E.cash
You can set some gradle properties and settings there and this cannot work if it's not a part of E.cash

It sounds counterintuitive to provide a dependency to a package you're installing but this addresses problems like 2 or more different version of AsyncStorage being installed

You need AsyncStorage in E.cash so you can configure it and link it to the app
As you've seen E.cash and Onyx actually declared different version of AsyncStorage
To avoid this problem only one package should install the dependency 
When `npm install` installs `react-native-onyx` it would hook it to the same version of AsyncStorage installed for the project

Usually libraries need to work together with other libraries that use the same packages, sometimes they even need to share the same instance of a module as well that's why they leave the project that installs them to resolve their peer dependencies

Reply:
Oh wow, that's really interesting. Funny that I've never ran into this before, but ReactNative is a bit of a different monster that I'm still getting used to. Thanks for the information!